templates/messageextension/csharp/default/Bot/TeamsMessageExtension.tpl

```diff
@@ -0,0 +1,210 @@
+﻿using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Teams;
+using Microsoft.Bot.Schema;
+using Microsoft.Bot.Schema.Teams;
+
+using Newtonsoft.Json.Linq;
+
+namespace {{BlazorAppServer}}.Bot;
+
+public class TeamsMessageExtension : TeamsActivityHandler
+{
+    // Message Extension Code
+    // Action.
+    protected override async Task<MessagingExtensionActionResponse> OnTeamsMessagingExtensionSubmitActionAsync(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionAction action, CancellationToken cancellationToken)
+    {
+        switch (action.CommandId)
+        {
+            case "createCard":
+                return CreateCardCommand(turnContext, action);
+            case "shareMessage":
+                return ShareMessageCommand(turnContext, action);
+        }
+        return new MessagingExtensionActionResponse();
+    }
+
+    private MessagingExtensionActionResponse CreateCardCommand(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionAction action)
```

Comment:
This looks similar to command bot scenario of "command handler" and "send response by card", so we could think about how to simplify it in the similar way, or perhaps leverage that middleware?